torch_geometric/data/hetero_data.py

```diff
@@ -616,6 +623,50 @@ def _consistent_size(stores: List[BaseStorage]) -> List[str]:
 
         return data
 
+    # :obj:`FeatureStore` interface ###########################################
+
+    def _put_tensor(self, tensor: FeatureTensorType, attr: TensorAttr) -> bool:
```

Comment:
Do we really want to support this `put_tensor` operation now? It looks like there are a lot of places could go wrong. What if index is out of bound (put new tensors)? What if `out` does not exist and `attr.index` is not continuous? What if `out` exists but `val` does not? I'd rather throw a not implemented error here.

Reply:
I think we should support `put_tensor` since all of the cases you mentioned (and others) are either covered by the function or error out in the same way that PyG would normally error out. In particular,

> What if index is out of bound (put new tensors)?

Not sure what you mean here. If we are putting a new tensor, then no node storage would be found and we would just store the tensor as intended. If you mean an index out of bounds exception for replacing a tensor, PyG would fail here as well (as expected)

> What if out does not exist and attr.index is not continuous? 

This is fine, we support any index that torch supports

> What if out exists but val does not?

Good point! Fixed.